mlir/include/mlir/Analysis/Presburger/IntegerRelation.h

```diff
@@ -711,6 +711,13 @@ class IntegerRelation {
   /// return `this \ set`.
   PresburgerRelation subtract(const PresburgerRelation &set) const;
 
+  // Remove equalities which have only zero coefficients.
+  void removeTrivialEqualities();
+
+  // Verify whether the relation is full-dimensional, i.e.,
+  // has the same number of dimensions as the number of variables.
```

Comment:
I don't think this is the correct definition of full dim. Doesn't it mean that no dimension is projected?

Reply:
As far as I could tell, this is how the term was used in Sven's paper. e.g.
> The regions in the parameter space Qn with a fixed set of parametric vertices, i.e., the chambers, form the elements of a subdivision of (part of) the parameter space. Any intersection of two elements of a subdivision is also part of the subdivision. We are only interested in the full-dimensional chambers here, i.e., those of dimension n.